pip/pip-433.md

```diff
@@ -0,0 +1,172 @@
+# PIP-433: Optimize the conflicts of the replication and automatic creation mechanisms, including the automatic creation of topics and schemas
+
+# Background knowledge
+
+#### Topic auto-creation by Rest API if you have enabled Geo Replication.
+
+The source broker will copy the REST requests that is a partitioned topic creation to the remote cluster if you have already enabled a namespace-level Geo Replication.
+
+The source broker will do the following things when you try to enable a topic-level Geo-Replication, and you did not enable a namespace-level Geo-Replication.
+- The source broker checks whether the partitioned topic exists under the remote cluster
+- If yes, the source broker compares both partition counts, enabling the topic-level replication if both clusters have the same number of partitions; otherwise, you will get a bad request error.
+- If not, it will create the partitioned topic with the same partitions count automatically under the remote cluster.
+
+#### Topic auto-creation by clients if you have enabled Geo Replication.
+- `Client of source cluster`: start a consumer/producer for a topic
+- `Client of source cluster`: Get partitions for topic
+    - At this step, the client will try to get the partitioned metadata of the topic. If the topic does not exist, the broker will create the partitioned metadata automatically.
+- `Client of source cluster`: Start internal consumers/producers for each sub-topic
+    - At this step, the client will try to connect to the sub-topic. If the sub-topic does not exist, the broker will create the partitioned metadata automatically.
```

Comment:
What does sub topic means here? internal partition of a partitioned topic?

Reply:
> What does sub topic means here? internal partition of a partitioned topic?

Yes, a internal partition of a partitioned topic

Reply:
Thanks for mentioning this, I added a description such as `sub-topic(an internal partition of a partitioned topic)`

Reply:
@poorbarcode We could use partition and partitioned topic directly instead of introducing new concept.